File: src/main/java/com/example/mapstruct/demo/EmployeeService.java
package com.example.mapstruct.demo;

import com.example.mapstruct.demo.controller.request.EmployeeRequest;
import com.example.mapstruct.demo.controller.response.EmployeesResponse;
import com.example.mapstruct.demo.entity.Employee;
import com.example.mapstruct.demo.entity.Employee_;
import lombok.RequiredArgsConstructor;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class EmployeeService {

    private final EmployeeRepository employeeRepository;
    private final EmployeeRequestResponseMapper employeeRequestResponseMapper;

    public List<EmployeesResponse> getEmployeesByFio(String fio) {
        List<Employee> employees = employeeRepository.findAll((Specification<Employee>) (root, query, criteriaBuilder)
                -> criteriaBuilder.equal(root.get(Employee_.FIO), fio));

        return employeeRequestResponseMapper.toResponse(employees);
    }

    public void add(EmployeeRequest employeeRequest) {
        Employee employee = employeeRequestResponseMapper.toEntity(employeeRequest);
        employeeRepository.save(employee);
    }
}

File: src/main/kotlin/com/example/mapstruct/demo/EmployeeService.kt
package com.example.mapstruct.demo

import com.example.mapstruct.demo.controller.request.EmployeeRequest
import com.example.mapstruct.demo.controller.response.EmployeesResponse
import com.example.mapstruct.demo.entity.Employee
import org.springframework.stereotype.Service

@Service
class EmployeeService(val employeeRepository: EmployeeRepository) {

    fun getEmployeesByFio(fio: String?): List<EmployeesResponse?>? {
        val employees: List<Employee?>? = employeeRepository.findByFio(fio)
        employees?.map { it?.toResponse() }?.toList()
        return employees?.map { it?.toResponse() }?.toList()
    }

    fun add(employeeRequest: EmployeeRequest?) {
        val employee = employeeRequest?.toEntity()
        employee?.let {
            employeeRepository.save(it)
        }
    }
}